srk-nixpkgs.nix

```diff
@@ -1,189 +1,54 @@
-    haskellPackageGen { } (pkgs.fetchFromGitHub {
-        owner = "input-output-hk";
-        repo = "cardano-report-server";
-        rev = "ede7a1a9eafd88e3449b38c7425f5529d6584014";
-        sha256 = "1j6dcjs80mxqq5r1p47c8076629hbsqw4kf63023s2rvw548qh4i";
-      })
-  ) { };
-
-  cardano-sl-core = hspkgs.callPackage (
-    haskellPackageGen { } ("${pkgs.fetchFromGitHub {
-        owner = "input-output-hk";
-        repo = "cardano-sl";
-        rev = "aa46ea4973b2afd3e1ec4d578cc10c1b63f5fcdb";
-        sha256 = "03cqs2ddlw1sgk5lbr19f2br7vn1nn22f3xibvarcscbmryjrp33";
-      }}/core")
-  ) { };
-  
-  cardano-sl = hspkgs.callPackage (
-    haskellPackageGen { } (pkgs.fetchFromGitHub {
-        owner = "input-output-hk";
-        repo = "cardano-sl";
-        rev = "aa46ea4973b2afd3e1ec4d578cc10c1b63f5fcdb";
-        sha256 = "03cqs2ddlw1sgk5lbr19f2br7vn1nn22f3xibvarcscbmryjrp33";
-      })
-  ) { rocksdb = rocksdb-haskell; };
-  
   hspkgs = compiler.override {
     overrides = self: super: {
-      inherit universum acid-state kademlia plutus-prototype node-sketch network-transport-tcp ed25519 log-warper cardano-report-server network-transport serokell-util cardano-sl-core;
-
-      QuickCheck = super.QuickCheck_2_9_2;
-      optparse-applicative = super.optparse-applicative_0_13_0_0;
-      criterion = super.criterion_1_1_4_0;
-      #code-page = super.code-page_0_1_1;
-
-      turtle = doJailbreak super.turtle_1_3_0;
-
-      cardano-sl = overrideCabal cardano-sl (drv: {
-        # Build statically to reduce closure size
-        enableSharedLibraries = false;
-        enableSharedExecutables = false;
-        isLibrary = false;
+      # To generate these go to ./pkgs and run ./generate.sh
+      universum = super.callPackage ./pkgs/universum.nix { };
+      serokell-util = super.callPackage ./pkgs/serokell-util.nix { };
+      acid-state = super.callPackage ./pkgs/acid-state.nix { };
+      log-warper = super.callPackage ./pkgs/log-warper.nix { };
+      ed25519 = super.callPackage ./pkgs/ed25519.nix { };
+      rocksdb = super.callPackage ./pkgs/rocksdb-haskell.nix { rocksdb = pkgs.rocksdb; };
+      kademlia = super.callPackage ./pkgs/kademlia.nix { };
+      node-sketch = super.callPackage ./pkgs/time-warp-nt.nix { };
+      cardano-report-server = super.callPackage ./pkgs/cardano-report-server.nix { };
+      cardano-crypto = super.callPackage ./pkgs/cardano-crypto.nix { };
+      plutus-prototype = super.callPackage ./pkgs/plutus-prototype.nix { };
+      network-transport = super.callPackage ./pkgs/network-transport.nix { };
+      network-transport-tcp = super.callPackage ./pkgs/network-transport-tcp.nix { };
+      derive = super.callPackage ./pkgs/derive.nix { };
+      cryptonite = super.callPackage ./pkgs/cryptonite.nix { };
+
+      # TODO: https://github.com/NixOS/cabal2nix/issues/261
+      cardano-sl-core = prodMode (super.callCabal2nix "cardano-sl-core" "${self.cardano-sl.src}/core" {});
```

Comment:
I didn't check, just got it to compile :) should we set it for all cardano-* packages?